src/pages/index.js

```diff
@@ -0,0 +1,319 @@
+  }
+
+  cardImage.addEventListener("click", () => {
+    previewImage.src = data.link;
+    previewImage.alt = data.name;
+    previewCaption.textContent = data.name;
+    openModal(previewModal);
+  });
+
+  cardLikeBtn.addEventListener("click", () => {
+    cardLikeBtn.classList.toggle("card__like-btn_liked");
+    handleCardLikeUnlike(cardLikeBtn, data);
+  });
+
+  cardDeleteBtn.addEventListener("click", () => {
+    handleDeleteCard(cardElement, data);
+  });
+
+  return cardElement;
+}
+
+function handleCardLikeUnlike(cardLikeBtn, data) {
+  selectedCardID = data._id;
+  if (cardLikeBtn.classList.contains("card__like-btn_liked")) {
+    api.addLike(selectedCardID).catch((err) => console.error(err));
+  } else {
+    api.removeLike(selectedCardID).catch((err) => console.error(err));
+  }
+}
+
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardID = data._id;
+  openModal(deletePostModal);
+}
+
+function handleDeleteSubmit() {
+  const originalText = deletePostBtn.textContent;
+  deletePostBtn.textContent = "Deleting...";
+  api
+    .removeCard(selectedCardID)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deletePostModal);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      deletePostBtn.textContent = originalText;
+    });
+}
+
+// === Event Listeners ===
+// Modal Open Events
+editProfileBtn.addEventListener("click", () => {
+  openModal(profileModal);
+  nameInput.value = profileNameElement.textContent;
+  jobInput.value = profileJobElement.textContent;
+});
+
+newPostBtn.addEventListener("click", () => openModal(newPostModal));
+profileEditImageBtn.addEventListener("click", () =>
+  openModal(profileEditImageModal)
+);
+
+// Modal Close Events
+closeProfileBtn.addEventListener("click", () => closeModal(profileModal));
+newPostCloseBtn.addEventListener("click", () => closeModal(newPostModal));
+deletePostBtn.addEventListener("click", handleDeleteSubmit);
+canelDeletePostBtn.addEventListener("click", () => closeModal(deletePostModal));
+deleteModalCloseBtn.addEventListener("click", () =>
+  closeModal(deletePostModal)
+);
+previewCloseBtn.addEventListener("click", () => closeModal(previewModal));
+profileImageEditCloseBtn.addEventListener("click", ()=> closeModal(profileEditImageModal));
+// Profile Form Submission
+profileFormElement.addEventListener("submit", handleProfileFormSubmit);
+function handleProfileFormSubmit(evt) {
+  evt.preventDefault();
+  const originalText = submitProfileBtn.textContent;
+  submitProfileBtn.textContent = "Saving...";
+  api
+    .editUserInfo({ name: nameInput.value, about: jobInput.value })
+    .then((data) => {
+      profileNameElement.textContent = data.name;
+      profileJobElement.textContent = data.about;
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      submitProfileBtn.textContent = originalText;
+    });
+
+  closeModal(profileModal);
```

Comment:
`closeModal(profileModal);`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**